modules/shared/server/controllers/uploader.controller.js

```diff
@@ -0,0 +1,170 @@
+const q = require('q');
+const multer = require('multer');
+const multerS3 = require('multer-s3');
+const aws = require('aws-sdk');
+const uuid = require('uuid');
+const path = require('path');
+const fs = require('fs');
+const confit = require(path.resolve('config/config'));
+
+function uploadController(logger) {
+  /**
+   * uploadLocal
+   * @param {Object} config - same as config object passed to upload
+   * @param {Object} config.local - strategy specific options
+   * @param {string} config.local.dest - directory to upload relative to project root
+   * @param {string} config.local.apiPrefix - api endpoint where file will be served from (no file name)
+   * @returns {Promise} will resolve to the file's url
+   */
+  function uploadLocal(config) {
+    let upload = multer({
+      storage: multer.diskStorage({
+        destination: (req, file, cb) => {
+          cb(null, config.local.dest);
+        },
+        filename: (req, file, cb) => {
+          cb(null, config.newFileName);
+        }
+      }),
+      fileFilter: config.fileFilter,
+      limits: config.local.limits
+    }).single('file');
+
+    let url = config.local.apiPrefix + config.newFileName;
+
+    return new Promise((resolve, reject) => {
+      upload(config.req, config.res, (err) => {
+        if (err) {
+          logger.error(err);
+          throw new Error('Error while uploading with local strategy');
+        } else {
+          resolve(url);
+        }
+      });
+    })
+    .then((newUrl) => {
+      return new Promise((resolve, reject) => {
+        if (config.oldFileName) {
+          fs.unlink(path.resolve(config.local.dest, oldFileName), (err) => {
+            if (err) {
+              logger.error('Error while deleting object locally', err);
+              throw err;
+            } else {
+              logger.debug('Old file deleted locally');
+              resolve(newUrl);
+            }
+          });
+        } else {
+          resolve(newUrl);
+        }
+      });
+    });
+  }
+
+  /**
+   * uploadS3
```

Comment:
FYI, These should be one line descriptions of what the function does.  Just the name is pretty useless